### PR TITLE
[Feature] Nested keys in `OrnsteinUhlenbeckProcess`

### PR DESCRIPTION
**Pull request by matteobettini**

### CODE REVIEW

File: test/test_env.py [hunk 1]
```diff
@@ -1368,22 +1368,22 @@ def test_mp_collector(self, nproc):
 
 
 class TestNestedSpecs:
-    @pytest.mark.parametrize("envclass", ["CountingEnv", "NestedRewardEnv"])
+    @pytest.mark.parametrize("envclass", ["CountingEnv", "NestedCountingEnv"])
     def test_nested_reward(self, envclass):
```
Comment (matteobettini) [inline test/test_env.py:1372]: Will need #1268 fixed to run

File: test/test_env.py [hunk 2]
```diff
@@ -1393,14 +1393,14 @@ def test_nested_reward(self, envclass):
...
-        if envclass == "NestedRewardEnv":
+        if envclass == "NestedCountingEnv":
             assert ("next", "data", "done") in next_state.keys(True)
             assert ("next", "data", "states") in next_state.keys(True)
             assert ("next", "data", "reward") in next_state.keys(True)
         assert ("next", *env.done_key) in next_state.keys(True)
         assert ("next", *env.reward_key) in next_state.keys(True)
 
-        check_env_specs(env)
+        # check_env_specs(env)
```
Comment (matteobettini) [inline test/test_env.py:1403]: Will need https://github.com/pytorch/rl/issues/1268 fixed to run

File: torchrl/modules/tensordict_module/exploration.py [hunk 3]
```diff
@@ -481,18 +483,19 @@ def step(self, frames: int = 1) -> None:
     def forward(self, tensordict: TensorDictBase) -> TensorDictBase:
         tensordict = super().forward(tensordict)
         if exploration_type() == ExplorationType.RANDOM or exploration_type() is None:
-            if "is_init" not in tensordict.keys():
+            if self.is_init_key not in tensordict.keys(True, True):
```
Comment (matteobettini) [inline torchrl/modules/tensordict_module/exploration.py:486]: I would like for the possibility of the init_key being nested. Is there a way we can run this check just once and then skip?

Comment (matteobettini) [reply torchrl/modules/tensordict_module/exploration.py]: I solved this

File: torchrl/modules/tensordict_module/exploration.py [hunk 4]
```diff
@@ -392,10 +392,11 @@ def __init__(
         x0: Optional[Union[torch.Tensor, np.ndarray]] = None,
         sigma_min: Optional[float] = None,
         n_steps_annealing: int = 1000,
-        action_key: str = "action",
+        action_key: Optional[NestedKey] = "action",
+        is_init_key: Optional[NestedKey] = "is_init",
```
Comment (vmoens) [inline torchrl/modules/tensordict_module/exploration.py:396]: seems to be missing from the docstrings

File: torchrl/modules/tensordict_module/exploration.py [hunk 5]
```diff
@@ -540,43 +545,69 @@ def noise_key(self):
...
         return self._steps_key  # + str(id(self))
 
-    def _make_noise_pair(self, tensordict: TensorDictBase, is_init=None) -> None:
+    def _make_noise_pair(
+        self,
+        action_tensordict: TensorDictBase,
+        tensordict: TensorDictBase,
+        is_init: torch.Tensor,
+    ):
+        if self.steps_key not in tensordict.keys():
```
Comment (vmoens) [inline torchrl/modules/tensordict_module/exploration.py:553]: can step_key be a tuple?

**Review by vmoens — APPROVED** [review]
LGTM thanks for this!

File: torchrl/modules/tensordict_module/exploration.py (hunk 5 above)
Comment (matteobettini) [reply torchrl/modules/tensordict_module/exploration.py:553]: not that key is not parametric

**Pull request merged**